### PR TITLE
if ratpack.server.ServerConfig#getAddress is null, set brave host/loc…

### PR DESCRIPTION
**Pull request by edwardsoo-ss**
…al endpoint Tp to first site-local java.net.NetworkInterface#getNetworkInterfaces instead of loopback

@llinder @Bijnagte @jeff-blaisdell please have a look

### CODE REVIEW

**Comment by llinder** [comment]
Though not much different than what you have, I'm little partial to the approach taken here: https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/main/java/zipkin/server/brave/BraveConfiguration.java#L66

It does have the advantage of working with ipv6 as well as falling back to loopback address if an exception is thrown.

We should also make this a pull request against https://github.com/hyleung/ratpack-zipkin if possible.  I think he would be completely open to this change.

File: src/main/java/ratpack/zipkin/ServerTracingModule.java [hunk 1]
```diff
@@ -115,6 +120,28 @@ else if (config.spanCollector != null) {
     return braveBuilder.build();
   }
 
+  private static byte[] getHostAddress() {
+    try {
+      return hostAddress().orElse(InetAddress.getLoopbackAddress()).getAddress();
+    } catch (SocketException e) {
+      return InetAddress.getLoopbackAddress().getAddress();
```
Comment (Bijnagte) [inline src/main/java/ratpack/zipkin/ServerTracingModule.java]: this looks a little goofy to have the `InetAddress.getLoopbackAddress()` repeated in both sides but it looks like it should work fine.

**Comment by edwardsoo-ss** [comment]
updated to work with ipv6 and call `InetAddress.getLoopbackAddress()` at most once

**Comment by Bijnagte** [comment]
👍

**Comment by llinder** [comment]
@edwardsoo-ss can you also open a pull request against https://github.com/hyleung/ratpack-zipkin with these changes?  We really need to look towards not maintaining a fork of this.

**Comment by edwardsoo-ss** [comment]
@llinder @Bijnagte https://github.com/hyleung/ratpack-zipkin/pull/21

**Pull request merged**